lib/src/validator/strict_dependencies.dart

```diff
@@ -35,59 +32,51 @@ class StrictDependenciesValidator extends Validator {
         directives = parseImportsAndExports(contents, name: file);
       } on AnalyzerErrorGroup catch (e, s) {
         // Ignore files that do not parse.
-        exception(e, s);
+        log.fine(getErrorMessage(e));
+        log.fine(new Chain.forTrace(s).terse);
         continue;
       }
       for (var directive in directives) {
         Uri uri;
         try {
           uri = Uri.parse(directive.uri.stringValue);
         } on FormatException catch (_){}
-        var usage = new _Usage(file, contents, directive, uri);
-        if (usage.isPackageUrl) {
+        // If the URL could not be parsed or it is a *package* AND
+        // there are no segments OR
+        // any segment are empty
+        if (uri == null ||
+            uri.scheme == 'package' &&
+            uri.pathSegments.length < 2 ||
+            uri.pathSegments.any((s) => s.isEmpty)) {
```

Comment:
Please parenthesize sub-expressions with different boolean operators. I'm sure the order of operation works out here, but it's a lot harder to visually verify that without explicit grouping.